src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -157,7 +157,8 @@ internal static DiagnosticInfo CreateObsoleteDiagnostic(Symbol symbol, BinderFla
             }
 
             // Issue a specialized diagnostic for add methods of collection initializers
-            var isColInit = location.Includes(BinderFlags.CollectionInitializerAddMethod);
+            var isColInit = location.Includes(BinderFlags.CollectionInitializerAddMethod) &&
+                            symbol.Name == WellKnownMemberNames.CollectionInitializerAddMethodName;
```

Comment:
>symbol.Name == WellKnownMemberNames.CollectionInitializerAddMethodName [](http://example.com/codeflow?start=28&length=70)

I think we should make sure that we are getting "correct" BinderFlags value instead. The value that we would get if we were using an object expression with explicit type. Probably we should save the Binder in ```BoundUnconvertedObjectCreationExpression``` node and use that binder to bind "conversion" to the target type (in ```ConvertObjectCreationExpression```).